### PR TITLE
Add prepend filters method

### PR DESCRIPTION
**Pull request by fink3l**
The pull request helps in situations when you collect part of the filters statically and some need to be collected dynamically and with a higher priority.

### CODE REVIEW

File: src/DeepCopy/DeepCopy.php [hunk 1]
```diff
@@ -85,11 +85,33 @@ public function copy($value)
         return $this->recursiveCopy($value);
     }
 
+    /**
+     * @param Filter  $filter
```
Comment (theofidry) [inline src/DeepCopy/DeepCopy.php]: I wouldn't add the phpdoc. When it doesn't provide any more information than the typehints (like here), it just adds noise IMO.

File: src/DeepCopy/DeepCopy.php [hunk 2]
```diff
@@ -85,11 +85,33 @@ public function copy($value)
...
+     */
     public function addFilter(Filter $filter, Matcher $matcher): void
     {
         $this->filters[] = [$matcher, $filter];
     }
 
+    /**
+     * Add filter with high priority
+     *
+     * @param Filter  $filter
```
Comment (theofidry) [inline src/DeepCopy/DeepCopy.php]: Same here. I think the method is self-describing enough to skip the description `Add filter with high priority`, but I don't mind leaving it either.

File: src/DeepCopy/DeepCopy.php [hunk 3]
```diff
@@ -85,11 +85,33 @@ public function copy($value)
...
+     * @param Matcher $matcher
+     *
+     * @return void
+     */
+    public function prependFilter(Filter $filter, Matcher $matcher): void
+    {
+        \array_unshift($this->filters, [
```
Comment (theofidry) [inline src/DeepCopy/DeepCopy.php]: I would rather add a `use function array_unshift;` statement at the top of the file than having this leading backslash. The leading backslash is necessary in older version of PHP as `use function` is not allowed there, but in this case we do not support those old versions. So it's more a matter of style and I find the use statement more readable

**Review by theofidry — COMMENTED** [review]
Thanks for the PR! There is some minor changes to do but looks good to me otherwise

File: src/DeepCopy/DeepCopy.php (hunk 1 above)
Comment (fink3l) [reply src/DeepCopy/DeepCopy.php]: i think you right, thanks

File: src/DeepCopy/DeepCopy.php (hunk 2 above)
Comment (fink3l) [reply src/DeepCopy/DeepCopy.php]: OK

File: src/DeepCopy/DeepCopy.php (hunk 3 above)
Comment (fink3l) [reply src/DeepCopy/DeepCopy.php]: I made mistake with branches, i think i made PR to 1.8. Damn, sorry for that.
I added unit tests and fix prependFilter method for 2.x version, now it's OK.

**Review by peter279k — APPROVED** [review]

**Comment by theofidry** [comment]
Thanks

**Pull request merged**